src/librustc_mir/dataflow/impls/borrows.rs

```diff
@@ -233,6 +233,21 @@ impl<'a, 'gcx, 'tcx> BitDenotation for Borrows<'a, 'gcx, 'tcx> {
     fn terminator_effect(&self,
                          sets: &mut BlockSets<BorrowIndex>,
                          location: Location) {
+        let block = &self.mir.basic_blocks().get(location.block).unwrap_or_else(|| {
+            panic!("could not find block at location {:?}", location);
+        });
+        match block.terminator().kind {
+            mir::TerminatorKind::Resume |
+            mir::TerminatorKind::Return |
+            mir::TerminatorKind::GeneratorDrop => {
+                for (borrow_index, borrow_data) in self.borrows.iter_enumerated() {
+                    if let ReScope(..) = borrow_data.region {
+                        sets.kill(&borrow_index);
+                    }
+                }
+            }
+            _ => {}
```

Comment:
Nit: I'd prefer to see the variants here written out exhaustively. My rule of thumb: if a new terminator kind is added, what is the change this code might want to change? If it is not "rather low", make it exhaustive. I rate this chance as "moderate".